contracts/manifold/collectible/ERC721Collectible.sol

```diff
@@ -10,308 +10,350 @@ import "@openzeppelin/contracts/utils/Strings.sol";
+    }
 
-    if (amount == 1) {
-      instance.purchaseCount++;
+    /**
+    * @dev See {IERC721CreatorExtensionApproveTransfer-setApproveTransfer}
+    */
+    function setApproveTransfer(
+      address creatorContractAddress,
+      bool enabled
+    ) external override creatorAdminRequired(creatorContractAddress) {
+        require(
+            ERC165Checker.supportsInterface(creatorContractAddress, type(IERC721CreatorCore).interfaceId),
+            "creator must implement IERC721CreatorCore"
+        );
+        IERC721CreatorCore(creatorContractAddress).setApproveTransferExtension(enabled);
+    }
 
-      // Mint token
-      uint256 tokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(to);
-      _tokenIdToTokenClaimMap[creatorContractAddress][tokenId] = TokenClaim(uint224(instanceId), instance.purchaseCount);
-      emit Unveil(creatorContractAddress, instanceId, instance.purchaseCount, tokenId);
-    } else {
-      uint32 tokenStart = instance.purchaseCount + 1;
-      instance.purchaseCount += amount;
+    /**
+    * @dev See {IERC721CreatorExtensionApproveTransfer-approveTransfer}.
+    */
+    function approveTransfer(address, address from, address, uint256 tokenId) external view override returns (bool) {
+        // always allow mints
+        if (from == address(0)) {
+            return true;
+        }
+        
+        uint256 instanceId = _tokenIdToTokenClaimMap[msg.sender][tokenId].instanceId;
+        require(instanceId != 0, "Token not found");
 
-      // Mint token
-      uint256[] memory tokenIds = IERC721CreatorCore(creatorContractAddress).mintExtensionBatch(to, amount);
+        CollectibleInstance storage instance = _getInstance(msg.sender, instanceId);
 
-      for (uint32 i = 0; i < tokenIds.length; ) {
-        emit Unveil(creatorContractAddress, instanceId, tokenStart + i, tokenIds[i]);
+        return !instance.isTransferLocked;
+    }
 
-        _tokenIdToTokenClaimMap[creatorContractAddress][tokenIds[i]] = TokenClaim(uint224(instanceId), tokenStart + i);
-        unchecked {
-          i++;
+    /**
+    * @dev override if you want to perform different mint functionality
+    */
+    function _mint(address creatorContractAddress, uint256 instanceId, address to, uint16 amount) internal virtual {
+        CollectibleInstance storage instance = _getInstance(creatorContractAddress, instanceId);
+
+        if (amount == 1) {
+            uint256 tokenId;
+            if (instance.contractVersion >= 3) {
+                uint80 tokenData = uint56(instanceId) << 24 | uint24(++instance.purchaseCount);
+                tokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(to, tokenData);
+            } else {
+                ++instance.purchaseCount;
+                tokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(to);
+                _tokenIdToTokenClaimMap[creatorContractAddress][tokenId] = TokenClaim(uint224(instanceId), instance.purchaseCount);
```

Comment:
I don't fully understand why we need the `TokenClaim` struct at all now (even for creator core version < 3) -- we could use the same trick for the token claim struct right? I guess the point is that doesn't save us anything since it'd still take up one slot? Still might be cleaner to save it as the same data vs a packed number in one instance and a struct in the other

Reply:
we can't, because you can mint non-sequentially.

Reply:
how could we use it properly if things were minted/claimed out of order?

Reply:
Oh, in version < 3, we didn't have the concept of uint80 token data